packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -1129,7 +1129,7 @@ class _AppBarState extends State<AppBar> {
           shadowColor: widget.shadowColor
             ?? appBarTheme.shadowColor
             ?? _defaultShadowColor,
-          shape: widget.shape,
+          shape: widget.shape ?? appBarTheme.shape,
```

Comment:
Shouldn't this behaviour be documented in the docs the `AppBar.shape` as well?

Reply:
Good point. I've updated the AppBar.shape API doc.